language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/CreateFunctionCodeAction.java

```diff
@@ -322,7 +322,9 @@ private Optional<NonTerminalNode> findEnclosingModulePartNode(NonTerminalNode no
     private boolean addNewLineAtEnd(Node enclosingNode) {
         Iterator<Node> iterator = enclosingNode.parent().children().iterator();
         while (iterator.hasNext()) {
-            if (iterator.next().lineRange().startLine().line() == enclosingNode.lineRange().endLine().line() + 1) {
+            Node next = iterator.next();
+            if (next.textRange().length() > 0
+                    && next.lineRange().startLine().line() == enclosingNode.lineRange().endLine().line() + 1) {
```

Comment:
Shall we move this method to CodeActionUtil?

Reply:
We can write this using a for loop also
```
    private boolean addNewLineAtEnd(Node enclosingNode) {
        for (Node child : enclosingNode.parent().children()) {
            if (child.textRange().length() > 0
                    && child.lineRange().startLine().line() == enclosingNode.lineRange().endLine().line() + 1) {
                return true;
            }
        }
        return false;
    }
    ```

Reply:
Using an iterator is best performance wise (faster)  than a for-loop. Hence, shall we keep this @gayalkuruppu